core/src/trezor/ui/layouts/tt_v2/webauthn.py

```diff
@@ -1,20 +1,73 @@
 from typing import TYPE_CHECKING
 
 from trezor import wire
+from trezor.enums import ButtonRequestType
 
-from ...components.common.webauthn import ConfirmInfo
+import trezorui2
+
+from ..common import interact
+from . import _RustLayout, is_confirmed
 
 if TYPE_CHECKING:
-    Pageable = object
+    from ...components.common.webauthn import ConfirmInfo
+    from ...components.tt.confirm import Pageable
 
 
 async def confirm_webauthn(
     ctx: wire.GenericContext | None,
     info: ConfirmInfo,
     pageable: Pageable | None = None,
-) -> bool:
-    raise NotImplementedError
+) -> tuple[bool, int]:
+    # `trezorui2.confirm_webauthn` is expecting a list of `app_names`
+    # and `account_names` - to allow for the `pageable` situation.
+    # In case the list has only one element, it will not be paginated.
+    # Assuming the header and icon are the same for all pages.
+
+    app_names: list[str] = [info.app_name()]
+    account_names: list[str | None] = [info.account_name()]
```

Comment:
Consider using [list of tuples](https://docs.python.org/3/library/functions.html#zip) - then you don't need to handle the case where lists are of different length and there will be slightly less asserts on the receiving side.

See also https://github.com/trezor/trezor-firmware/blob/da83d985392d45eb5e3bb15cd2ff19692babf46a/core/embed/rust/src/ui/model_tt/layout.rs#L1064

Reply:
Done in [b557d76](https://github.com/trezor/trezor-firmware/pull/2556/commits/b557d76468ea90567cba981b4288e96502a5b73f)